### PR TITLE
Fix: Verification of checksum from file attr

### PR DESCRIPTION
**Pull request by jrohel**
Related to https://bugzilla.redhat.com/show_bug.cgi?id=1700341

File copy could result in change in file attributes where
null-terminators are stripped out. The new code does not relly on it.

### CODE REVIEW

**Comment by j-mracek** [comment]
:+1: 
LGTM

File: librepo/checksum.c [hunk 1]
```diff
@@ -221,18 +221,18 @@ lr_checksum_fd_compare(LrChecksumType type,
...
-            attr_ret = fgetxattr(fd, key, &buf, 256);
-            if (attr_ret != -1) {
+            ssize_t attr_size = fgetxattr(fd, key, &buf, sizeof(buf));
+            if (attr_size != -1) {
                 // Cached checksum found
                 g_debug("%s: Using checksum cached in xattr: [%s] %s",
                         __func__, key, buf);
-                *matches = strcmp(expected, buf) ? FALSE : TRUE;
+                size_t expected_len = strlen(expected);
+                *matches = expected_len > attr_size || memcmp(expected, buf, expected_len) ? FALSE : TRUE;
```
Comment (m-blaha) [inline librepo/checksum.c]: But this would return true whenever attribute value read from the filesystem starts with expected value. Isn't that too soft?

Comment (j-mracek) [reply librepo/checksum.c]: According to discussion with jrohel it is evaluated like:
(expected_len > attr_size || memcmp(expected, buf, expected_len)) ? FALSE : TRUE;
if expected_len > attr_size or data differs it returns FALSE.

Comment (m-blaha) [reply librepo/checksum.c]: I meant this situation:
expected='AB'
buf='ABCD'
Wouldn't it return TRUE then?

Comment (j-mracek) [reply librepo/checksum.c]: Yes, but the checksum has always the same length, therefore there is no chance of that situation. The problem here is that fgetxattr could theoretically return checksum with or without null character on the end (null character is not kept after file copy) therefore comparison of size could fail for no good real reason.

Comment (jrohel) [reply librepo/checksum.c]: @m-blaha 
Actual state:
The "expected_len" is 64 (sha256 encoded in hexadecimal characters string). The "attr_size" is 64 or 65 (depends if null terminator is present in attr).
But I agree with you. It is soft for general usage. I will change the condition to be more strict.

**Comment by jrohel** [comment]
@m-blaha The comparison is strict now.

**Comment by m-blaha** [comment]
@rh-atomic-bot r+

**Pull request merged**